apiserver/facades/client/spaces/opfactory.go

```diff
@@ -16,7 +17,12 @@ type OpFactory interface {
 	NewRemoveSpaceModelOp(fromName string) (state.ModelOperation, error)
 
 	// NewRenameSpaceModelOp returns an operation for renaming a space.
-	NewRenameSpaceModelOp(fromName, toName string) (state.ModelOperation, error)
+	NewRenameSpaceModelOp(spaceName, toName string) (state.ModelOperation, error)
```

Comment:
`...(state.ModelOperation, error)` - We should really return the struct here.

Reply:
In this case we can't indirect the factory for testing if we do that.

Reply:
Yeah, you'll end up with shims if `state.ModelOperation` becomes more complicated. That's the only issue I generally hit.